lib/cluster/index.ts

```diff
@@ -212,7 +212,7 @@ class Cluster extends EventEmitter {
     if (!this.manuallyClosing && typeof this.options.clusterRetryStrategy === 'function') {
       retryDelay = this.options.clusterRetryStrategy.call(this, ++this.retryAttempts, reason)
     }
-    if (typeof retryDelay === 'number') {
+    if (typeof retryDelay === 'number' && retryDelay !== NaN) {
```

Comment:
it can be replaced with `if (isNaN(retryDelay))`

Reply:
well this could also be solved with simple retryDelay >= 0.  Now negative number just keeps retrying.

Reply:
I'm wondering what's the point of this change. Could you please provide an example to demonstrate in which case this change helps?

Reply:
clusterRetryStrategy: (times) => {
                    if (times > 3) { // stop every 3 times, retry next time.
                        return;
                    }
                    return Math.min(100 + times * 2, 1000);
                }

with this retry strategy and strict ts_config I get following error:

 Argument of type '{ clusterRetryStrategy: (times: number) => number | undefined; red
isOptions: { maxRetriesPerRequest: number; connectTimeout: number; lazyConnect: false; reconnectOnError: (err: Error) =
> false | 2; }; scaleReads: string; }' is not assignable to parameter of type 'ClusterOptions'.
  Types of property 'clusterRetryStrategy' are incompatible.
    Type '(times: number) => number | undefined' is not assignable to type '(times: number) => number'.
      Type 'number | undefined' is not assignable to type 'number'.
        Type 'undefined' is not assignable to type 'number'.

Reply:
Sorry for the late response. @types/ioredis module has been updated to support returning a `null` to stop reconnection, thus I think this pull request can be closed.